runners/flink/src/test/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslationsTest.java

```diff
@@ -122,4 +156,206 @@ public void testBeamKinesisSchemaFutureOccurredAtTimestamp() throws Exception {
       return bos.toByteArray();
     }
   }
+
+  @Test
+  public void testRequiredPropertiesForKafkaInput() throws JsonProcessingException {
+
+    LyftFlinkStreamingPortableTranslations portableTranslations =
+        new LyftFlinkStreamingPortableTranslations();
+    String id = "1";
+    String topicName = "kinesis_to_kafka";
+
+    Properties kafkaConsumerProps = new Properties();
+    kafkaConsumerProps.put("group.id", "testing");
+    kafkaConsumerProps.put("bootstrap.servers", "test-hdd.lyft.com");
+
+    ImmutableMap<String, Object> payload =
+        ImmutableMap.<String, Object>builder()
+            .put("topic", topicName)
+            .put("properties", kafkaConsumerProps)
+            .put("username", "read_kinesis_to_kafka")
+            .put("password", "abcde1234")
+            .build();
+
+    byte[] payloadBytes = new ObjectMapper().writeValueAsBytes(payload);
+    RunnerApi.Pipeline pipeline = createPipeline(id, payloadBytes);
+
+    portableTranslations.translateKafkaInput(id, pipeline, streamingContext);
+
+    ArgumentCaptor<FlinkKafkaConsumer011> kafkaSourceCaptor =
+        ArgumentCaptor.forClass(FlinkKafkaConsumer011.class);
+    ArgumentCaptor<String> kafkaSourceNameCaptor = ArgumentCaptor.forClass(String.class);
+    verify(streamingEnvironment)
+        .addSource(kafkaSourceCaptor.capture(), kafkaSourceNameCaptor.capture());
+    Assert.assertEquals(
+        WindowedValue.class, kafkaSourceCaptor.getValue().getProducedType().getTypeClass());
+    Assert.assertTrue(kafkaSourceNameCaptor.getValue().contains(topicName));
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testMissingRequiredCredentialsForKafkaInput() throws JsonProcessingException {
+
+    LyftFlinkStreamingPortableTranslations portableTranslations =
+        new LyftFlinkStreamingPortableTranslations();
+    String id = "1";
+    String topicName = "kinesis_to_kafka";
+
+    Properties kafkaConsumerProps = new Properties();
+    kafkaConsumerProps.put("group.id", "testing");
+    kafkaConsumerProps.put("bootstrap.servers", "test-hdd.lyft.com");
+
+    ImmutableMap<String, Object> payload =
+        ImmutableMap.<String, Object>builder()
+            .put("topic", topicName)
+            .put("properties", kafkaConsumerProps)
+            .build();
+
+    byte[] payloadBytes = new ObjectMapper().writeValueAsBytes(payload);
+    RunnerApi.Pipeline pipeline = createPipeline(id, payloadBytes);
+
+    portableTranslations.translateKafkaInput(id, pipeline, streamingContext);
+    Assert.fail("Should fail due to empty username and password! ");
+  }
+
+  @Test
+  public void testKafkaInputForLocalEnvironment() throws JsonProcessingException {
```

Comment:
I don't see much added value in this test to be honest. It seems to check the same as `testRequiredPropertiesForKafkaInput`. At least, the duplicate test code could be factored out.